### DISCUSSION TITLE
brew install vulnerable 7-zip version

### DISCUSSION THREAD

**Discussion by jpstotz** [Homebrew/discussions#1029] [Everyday usage]
`brew install p7zip` installs 7-zip version 17.03 which is to my knowledge vulnerable to some of the attacks mentioned here:
https://www.cvedetails.com/vulnerability-list/vendor_id-9220/7-zip.html

Please update or remove it.

**Comment by SMillerDev** [comment]
There isn't a newer release so there is nothing to update to. But it seems that you are wrong anyway: https://github.com/jinfeihan57/p7zip/releases/tag/v17.01

**Reply by jpstotz** [reply]
Thanks for clarification. 

The author of p7zip did a great job in confusing people especially because p7zip shows the following header info when started:

```
7-Zip [64] 17.03 : Copyright (c) 1999-2020 Igor Pavlov : 2017-08-28
p7zip Version 17.03 (locale=utf8,Utf16=on,HugeFiles=on,64 bits,8 CPUs x64)
```

The only reliable info source seems to be the command `brew info p7zip` that shows the link you have included in your post. 

But in the end considering that it took up to four years to fix those issues I will uninstall this piece of software and I would recommend it to other to do the same as this is not a well-maintained project.

**Reply by Bo98** [reply]
I think what happened is the original project (https://sourceforge.net/projects/p7zip/) became unmaintained. Someone then decided to fork the project in early 2020 to fix those exploits, so it seems better maintained now (or at least enough to make us switch the formula to use that version instead of patching the original).

**Reply by jpstotz** [reply]
It seems like the [p]7zip problem may be solved soon: 

7zip (the original version not p7zip) seems to get official Linux support: https://sourceforge.net/p/sevenzip/discussion/45797/thread/cec5e63147/

I don't know how much is left to port a Linux command-line app to MacOS, but I would assume that the necessary changes will not be that large.
